### PR TITLE
Merge feature/openapi/final into Main

### PR DESCRIPTION
**Pull request by nhtruong**
## Switching over to native OpenAPI:
- Removed Smithy:
  - Removed Smithy models (`model` folder)
  - Removed Gradle
  - Removed legacy test framework
  - Removed build-openapi-specs workflow
- Added Editable OpenAPI spec:
  - Added multi-file OpenApi spec (`spec` folder)
  - Added a tool to merge multi-file spec into single-file spec
  - Added build-single-file-specs workflow
- Updated Documents
## Adding Request and Response Bodies

The OpenAPI spec has also been updated with the request and response bodies that we extracted from ElasticSearch OpenAPI spec. This is done within the same PR as the native OpenAPI switch because we also used the ElasticSearch OpenAPI spec to map the query/path schemas to the schemas of many components in the bodies.

---

### Reviewers: It's easier to review this PR like reviewing a new repo by inspecting the [feature branch](https://github.com/opensearch-project/opensearch-api-specification/tree/feature/openapi/final)

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# TODO: Update this workflow API coverage tool is ready to handle yaml files
```
Comment (dblock) [inline .github/workflows/coverage.yml:1]: I think you want  https://github.com/opensearch-project/opensearch-api-specification/pull/195

File: tools/merger/merge.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import OpenApiMerger from "./OpenApiMerger";
+
+
+async function main() {
+    const merger = new OpenApiMerger('../spec/OpenSearch.openapi.yaml');
+    merger.merge('../builds/OpenSearch.latest.yaml');
+}
```
Comment (dblock) [inline tools/merger/merge.ts:7]: Tools should take source and destination as parameters rather than hard-code for this repo.

**Review by dblock — APPROVED** [review]
Looked quickly over the code and the specs, looks good! I'd like to discuss checking in the output spec vs. publishing it as a draft "latest" release, but we can do that later.

**Comment by dblock** [comment]
@Xtansia take a look, merge?

**Review by Xtansia — CHANGES_REQUESTED** [review]
Everything looks good generally, I have a similar concern to @dblock regarding checking in the built spec, but agree it can wait until after. Only issue I noticed was that there's a couple remaining references to xpack that should be tidied up

File: spec/schemas/nodes.info.yaml [hunk 3]
```diff
@@ -0,0 +1,763 @@
...
+        http:
+          $ref: '#/components/schemas/NodeInfoSettingsHttp'
+        bootstrap:
+          $ref: '#/components/schemas/NodeInfoBootstrap'
+        transport:
+          $ref: '#/components/schemas/NodeInfoSettingsTransport'
+        network:
+          $ref: '#/components/schemas/NodeInfoSettingsNetwork'
+        xpack:
+          $ref: '#/components/schemas/NodeInfoXpack'
```
[the remark this answers is not in the document]
Comment (nhtruong) [reply spec/schemas/nodes.info.yaml:327]: Thanks for catching that. I'll update the code to include this exception and regenerate the open API spec.

[the remark this answers is not in the document]
Comment (nhtruong) [reply spec/schemas/nodes.info.yaml]: https://github.com/opensearch-project/opensearch-api-specification/pull/208/commits/fd35aead0517f5df8b94dc9f1ff2817040135b7e

File: tools/merger/merge.ts (hunk 2 above)
Comment (nhtruong) [reply tools/merger/merge.ts:9]: Resolved by https://github.com/opensearch-project/opensearch-api-specification/pull/208/commits/0958319e2eca591c3d3f356413b045f78b2785c0

**Comment by nhtruong** [comment]
@dblock the API Coverage workflow is still being triggered from main branch. We will have to disable the workflow in the repo settings to get this merged.

**Comment by dblock** [comment]
> @dblock the API Coverage workflow is still being triggered from main branch. We will have to disable the workflow in the repo settings to get this merged.

That's ok.

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@Xtansia good with you?

**Pull request merged**